### PR TITLE
Honor CARES_ROOT_DIR

### PR DESCRIPTION
**Pull request by KyleFromKitware**
Honor CARES_ROOT_DIR in the add_subdirectory() call.

Fixes: #17876

@yang-g
@zackgalbreath

### CODE REVIEW

File: cmake/cares.cmake [hunk 1]
```diff
@@ -22,7 +22,7 @@ if("${gRPC_CARES_PROVIDER}" STREQUAL "module")
     # See https://github.com/grpc/grpc/issues/17255
     set(HAVE_LIBNSL OFF CACHE BOOL "avoid cares dependency on libnsl")
   endif()
-  add_subdirectory(third_party/cares/cares)
+  add_subdirectory("${CARES_ROOT_DIR}" third_party/cares/cares)
```
Comment (jtattermusch) [inline cmake/cares.cmake]: Looking at the original issue again, I'm not 100% sure what's the use of supporting CARES_ROOT_DIR.
The gRPC_CARES_PROVIDER=module branch is for enabling developer builds with the git repo checked out (and with all the submodules present).

I'd understand if we wanted to provide support for CARES_ROOT_DIR   when  gRPC_CARES_PROVIDER=package (that's the mode which is used when installing grpc and it makes sent to allow user pointing to a specific place where CARES is installed.

Comment (jtattermusch) [reply cmake/cares.cmake]: That said, it seems that we do support this for other dependencies (so we could add for consistency?):
https://github.com/grpc/grpc/blob/933acd31a8c06e09af495af1edc80645041f0d27/cmake/ssl.cmake#L16

https://github.com/grpc/grpc/blob/933acd31a8c06e09af495af1edc80645041f0d27/cmake/protobuf.cmake#L27

https://github.com/grpc/grpc/blob/933acd31a8c06e09af495af1edc80645041f0d27/cmake/zlib.cmake#L16

If we do this, let's stay consistent with how it's done in the other .cmake files?

Comment (KyleFromKitware) [reply cmake/cares.cmake]: "Consistent" meaning "without the quotes around `{CARES_ROOT_DIR}`"?

Comment (jtattermusch) [reply cmake/cares.cmake]: Sorry, misread the code.

**Comment by jtattermusch** [comment]
Known test failures:
https://github.com/grpc/grpc/issues/20703
https://github.com/grpc/grpc/issues/19619
https://github.com/grpc/grpc/issues/20545

**Pull request merged**